CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsChannelAccountEx.cs

```diff
@@ -0,0 +1,14 @@
+namespace Microsoft.Bot.Connector.Teams.Models
+{
+    /// <summary>
+    /// Teams channel account detailing user Azure Active Directory details.
+    /// </summary>
+    public partial class TeamsChannelAccount
+    {
+        /// <summary>
+        /// Sets the Azure Active Directory object Id from the other JSON field it can appear in.
+        /// </summary>
+        [Newtonsoft.Json.JsonProperty(PropertyName = "aadObjectId")]
+        private string AADObjectId { set { ObjectId = value; } }
+    }
```

Comment:
Use Swagger generation for generating this file. Also JS might need the same changes.

Reply:
I don't want to expose this as another property on the object. Its just there because sometimes we return the AAD object Id as aadObjectId or objectId.

Want to keep this private and to just set the ObjectId prop.

Reply:
I don't see the correlated place to put this into the node library.

Reply:
The AAD Object Id comes from:
- the field 'objectId' if fetched from the roster API
- the field 'aadObjectId' if passed in through a message from bot framework.

Reply:
Lets have a bigger discussion around this internally and bring both in sync it just seems wrong that we are passing around 2 values.

Reply:
It is wrong but we can't break legacy reliance on these fields either so I think we're stuck with them. We can add objectId AND aadObjectId to both situations and then refactor this code but for now this is the best we got.

Reply:
By adding this we will be building that legacy. We are only handling objectId for now so it is fine saying that aadObjectId was a bug. We need to close on this and then if we decide we can merge this PR as well. But we need a concrete decision first.

Reply:
<b>aadObjectId was a bug</b>might be true, however it is a bug that we have multiple partners relying on already so we cannot remove it or rename it. We can add ObjectId without removing aadObjectId to the smba responses and only after that, can we refactor this fix. Until then this is the best we have to make developer's lives easier. Right now, the only way to reliably get the AAD object Id is to have a workaround like this:
```c#
var aadId = (string)account.Properties["aadObjectId"] ?? (string)account.Properties["objectId"];
```
If we check this in users can just call AsTeamsChannelAccount() and expect to always get the AAD Object Id and once we add objectId to SMBA responses, we can refactor this to only read that property.